### ISSUE TITLE
Feature request - clipTo / crop opacity param

### ISSUE THREAD

**Issue by c-gross** [kangax/fabric.js#4605]
Instead of just having a mask (clipto / crop) were everything outside the mask is gone, it would be great if we could set a opacity value.

I know there is globalCompositeOperation but that doesn't work with one global mask and multiple canvas objects.

**Comment by asturur** [comment]
Well clipping is clipping, fabricjs offers higher level functionalities on top of canvas basic api.
But something related to clip has to hard cut everything that is outside, or should be named differently.

If you are looking at something like this: #4606 is not something that clipping should take care of.

Or better state you feature idea.

**Comment by c-gross** [comment]
Yes it's true it should be a different method and not in clipping. Something that does not cut everything outside completely.

Is this something that could be added as a new feature? If not, do you have an idea what would be the best approach?

**Comment by asturur** [comment]
this would just be a mask i think, a feature we do not have. Also we should see after clipping paths are ready because they could work also like masks.
